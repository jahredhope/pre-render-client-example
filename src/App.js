import React, { Component } from 'react';

import { injectGlobal } from 'emotion';
// import styled from 'react-emotion';

import Card from './components/Card';
import Text from './components/Text';

import reset from './reset';

reset();

// eslint-disable-next-line no-unused-expressions
injectGlobal`
  body {
    background-color: hsl(190, 5%, 96%);
    font-family: Helvetica,Arial,sans-serif;
  }
`;

export default class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      timeRendered: undefined
    };
  }
  componentDidMount() {
    this.updateTimeRendered();
  }
  updateTimeRendered = () => {
    this.setState({
      timeRendered: new Date().toString()
    });
    setTimeout(this.updateTimeRendered, 1000);
  };
  render() {
    return (
      <div>
        <Card>
          <Text heading>Base Application</Text>
          <Text>A simple boilerplate application</Text>
        </Card>
        {this.state.timeRendered && (
          <Card transparent>
            <br />
            <Text>{this.state.timeRendered}</Text>
          </Card>
        )}
      </div>
    );
  }
}
