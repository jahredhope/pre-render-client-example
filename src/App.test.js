import React from 'react';
import App from './App';
import { shallow } from 'enzyme';

jest.useFakeTimers();

describe('App', () => {
  it('should render', () => {
    expect(shallow(<App />)).toMatchSnapshot();
  });
});
