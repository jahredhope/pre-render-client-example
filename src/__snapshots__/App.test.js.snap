// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render 1`] = `
<div>
  <Styled(div)>
    <Styled(div)
      heading={true}
    >
      Base Application
    </Styled(div)>
    <Styled(div)>
      A simple boilerplate application
    </Styled(div)>
  </Styled(div)>
  <Styled(div)
    transparent={true}
  >
    <br />
    <Styled(div)>
      Sun Jan 01 2017 23:00:00 GMT+1100 (AEDT)
    </Styled(div)>
  </Styled(div)>
</div>
`;
